runtime/vm/jvminit.c

```diff
@@ -2421,6 +2426,16 @@ static UDATA checkArgsConsumed(J9PortLibrary* portLibrary, J9VMInitArgs* j9vm_ar
 	return TRUE;
 }
 
+/* Returns TRUE if a string is empty or if it contains only whitespace characters */
+static BOOLEAN isEmpty(const char * str) {
+	while('\0' != *str) {
+		if (0 == isspace((unsigned char) *str)) {
```

Comment:
@JasonFengJ9 great question - I wasn't entirely sure about this, but here is my thinking:
`isspace(int c)` expects that c is representable as an `unsigned char` or is equal to `EOF`, otherwise the behaviour is undefined. I read that C/C++ accepts `char` to be signed or unsigned, depending on the platform and compiler. To ensure that `isspace()` receives an `unsigned char`, I added the cast. Should we expect to deal with either signed or unsigned `char`? If we know that we will only deal with unsigned, I can remove the cast.

Reply:
@sharon-wang good thinking! this `optionString` could be from outside of `JVM` just like the legacy application #5332, so technically it might contain signed char values which are not `ASCII` codes. 
It appears this PR is going to reject such `optionString` with `JVMJ9VM007E Command-line option unrecognised`, what's the `RI` behaviors?

Reply:
@JasonFengJ9 do you have it backwards? OpenJ9 already rejects empty `optionString` with `JVMJ9VM007E Command-line option unrecognised`. This change is to ignore the empty options, to match the RI behavior.

Reply:
@pshipton right, this PR intends to ignore the empty options to match `RI` behaviors. From code perspective particularly inspired by this `unsigned char` cast, this PR is still going to reject those `signed char` options, my question was: does `RI` reject them as well or still ignore?

Reply:
@JasonFengJ9 Compared behaviour between RI and OpenJ9 - RI ignores and this current implementation does in fact reject.

To deal with `signed char`, a check can be added to this function so that we break and return false if we encounter one. With this change, it would probably be better to rename this function to reflect that we are checking if we should ignore, rather than just an empty check. Would this be a reasonable approach?

Reply:
Chatted with @sharon-wang about this, we might need some help from JCL team to give a high level description about what `char` does `RI` ignore or reject.

Reply:
I don't think we should be ignoring `signed chars`, i.e. chars > 0x7fff, if that is what you mean.

Reply:
Apparently `RI` ignores some `char` values `>0x7F`, need some hints from `JCL` to match `RI` behaviors.

Reply:
What are you testing? It wouldn't be hard to use glaunch or some custom launcher and run 255 times, passing in each single char value to check the behavior. Although according to the spec, the options are originally passed in the platform encoding, so that would only tell us the behavior for the current encoding. By the time we get here, I'd expect the options to be in UTF8, although I've only found code for Windows that does the conversion.

Reply:
Testing all possible `char` values was suggested to Sharon initially though I felt it is kind of hack. In addition I was wondering if there is any possibility that a `char` string pattern is to be checked instead of just single `char` value. That's why I suggested to seek JCL advice for accurate `RI` behaviours description. 

As per https://github.com/eclipse/openj9/files/3035727/sampleJNI.zip, the optionString was passed in within following code snippet:
```
       JavaVMOption* options = new JavaVMOption[2];   // JVM invocation options
       options[0].optionString = "";     ---> this is what JVM received.
```
A non-ascii code could be passed as well like following:
```
char testString[10];
testString[0] = 0xA0;
testString[1] = 0x0;
options[0].optionString = testString;
```
It appears `RI` ignore this `optionString` as well but this PR doesn't. So we are looking for a better understanding about what can be ignored or rejected by `RI`.

Reply:
I don't think we need the cast as `isSpace` is declared to take an `int`.  Both signed and unsigned chars can be promoted to int.

Further, `isSpace(int)` handles the most likely space characters and using it allows this PR to solve the problem originally reported.  Let's not rathole down every possible character combination right now.  It's fine to follow up with the JCL team and target matching the RI for those additional cases in a future release.

Let's get this fix into master and the 0.14 branch if we can.